msdk-featdet/msdk-featdet-openms/src/main/java/io/github/msdk/featdet/openms/FeatureFinderMetaboDetector.java

```diff
@@ -43,128 +43,129 @@
-   */
-  private String execShellCommand(String cmd) throws MSDKException {
-    String out = "";
-    try {
-      Process cmdProc = Runtime.getRuntime().exec(cmd);
-      BufferedReader stdoutReader =
-          new BufferedReader(new InputStreamReader(cmdProc.getInputStream()));
-      String line;
-      while ((line = stdoutReader.readLine()) != null) {
-        out += line + "\n";
-      }
-    } catch (IOException e) {
-      throw new MSDKException(e.getMessage());
-    }
-    return out;
-  }
+	private final String OPENMS_FEATURE_FINDER_METABO_LIBRARY_NAME;
+	private final @Nonnull File FEATURE_XML_FILE;
+	private final @Nonnull String FEATURE_XML_FILE_PATH;
+
+	private final @Nonnull File mzMLFile;
+	private final @Nonnull String mzMLFILE_PATH;
+
+	/**
+	 * 
+	 * <p>
+	 * Constructor for FeatureFinderMetaboDetector.
+	 * </p>
+	 * 
+	 * @param mzMLFilePath
+	 *            Path to mzML File to be processed
+	 */
+	public FeatureFinderMetaboDetector(String mzMLFilePath) throws MSDKException {
+		this(new File(mzMLFilePath));
+	}
+
+	/**
+	 * 
+	 * <p>
+	 * Constructor for FeatureFinderMetaboDetector.
+	 * </p>
+	 * 
+	 * @param mzMLFile
+	 *            {@link java.io.File} reference to mzML File to be processed
+	 */
+	public FeatureFinderMetaboDetector(File mzMLFile) throws MSDKException {
+		this.mzMLFile = mzMLFile;
+		FEATURE_XML_FILE = new File(System.getProperty("java.io.tmpdir") + UUID.randomUUID() + ".featureXML");
+		OPENMS_FEATURE_FINDER_METABO_LIBRARY_NAME = FeatureFinderMetaboLocator.findFeatureFinderMetabo();
+		mzMLFILE_PATH = mzMLFile.getAbsolutePath();
+		FEATURE_XML_FILE_PATH = FEATURE_XML_FILE.getAbsolutePath();
+	}
+
+	/**
+	 * Runs FeatureFinderMetabo on the output file, processes the output file
+	 * for mz values data.
+	 * 
+	 * @return List of m/z Values
+	 * @throws XMLStreamException
+	 * @throws IOException
+	 * @throws MSDKException
+	 */
+	public List<Double> execute() throws XMLStreamException, IOException, MSDKException {
+		List<Double> mzValuesResult = new ArrayList<Double>();
+		createTempFeatureXMLFile();
+
+		XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+		XMLEventReader xmlEventReader = xmlInputFactory.createXMLEventReader(new FileReader(FEATURE_XML_FILE));
+
+		while (xmlEventReader.hasNext()) {
+			XMLEvent xmlEvent = xmlEventReader.nextEvent();
+			if (xmlEvent.isStartElement()) {
+				StartElement startElement = xmlEvent.asStartElement();
+				if (startElement.getName().getLocalPart().equals("position")) {
+					xmlEvent = xmlEventReader.nextEvent();
+					Attribute idAttr = startElement.getAttributeByName(new QName("dim"));
+					if (idAttr.getValue().equals("1")) {
+						Characters characters = xmlEvent.asCharacters();
+						/*
+						 * Stores 1573.19798906671 in the ArrayList from the XML
+						 * line: <position dim="1">1573.19798906671</position>
+						 */
+						mzValuesResult.add(Double.valueOf(characters.getData()));
```

Comment:
This part needs some exception handling - what if the text contains something that cannot be parsed into a Double?